pkg/api/controller.go

```diff
@@ -1946,6 +1947,13 @@ func (c *Controller) CreateRepository(w http.ResponseWriter, r *http.Request, bo
 		return
 	}
 
+	if !c.Config.Installation.AllowInterRegionStorage {
+		if err := block.ValidateInterRegionStorage(r.Context(), c.BlockAdapter, body.StorageNamespace); err != nil {
```

Comment:
According to @arielshaqed 's suggestion, extracted this into a separate file.
It encapsulates it from this huge file, and make it easier to test.